Recipe page - Jasmin

JasminHed opened the pull request:
Hello, thanks for looking at my code! 

https://jrecipelibrary.netlify.app/

Commit:
added basic structure html and css file

Commit:
updated buttons, color

Commit:
updated id in html, css and variabel in javascript file

Commit:
updated id message in html and function in javascript

Commit:
updated flexbox with parent and mediaqueries

Commit:
updated mediaquieres and finalized javascript

Commit:
updated comments in js, mediaqueri for tablet and removed a styling element

Commit:
updated js with old code

Commit:
updated comments in js and put p in messages

Commit:
cleaned cond.statements from js

Commit:
removed recipe item in html, added radio buttons, updated styling for buttons, mediaquery tablet and desktop, added new javascript recipe array+function=dynamic content to DOM

Commit:
updated css styling in mobile, tablet and desktop view

Commit:
removed parent2, left recipe-container, updated tablet mediaquere, left parent2 in css

Commit:
commented out parent2, test

Commit:
updated mediaqueries for tablet, center and margin l+r in desktop

Commit:
added js function to make it get my alternatives or value set on my radiobuttons, eg vegan

Commit:
added an eventlistener so that when diet radiobutton is clicked, it will show the diet, eg button-vegan=vegan in console

Commit:
added js if click all-button=all recipes should show. Not working

Commit:
updated js due to error message filterValue has no value. Fixed w. if statement now inside function

Commit:
removed hard brackets, array, from diet name in mockup recipe list

Commit:
removed join from value recipe diet, not an array anymore

Commit:
updated mockup recipe so diet just have 1 string/diet

Commit:
added tolowercase so my values match, also changed nonveg to non-veg to match

Commit:
js getting my sort time value with functions

Commit:
added if, else recipes sorted on time

Commit:
added eventlistener to sort radio buttons

Commit:
added loadRecipes

Commit:
added gap in recipecontainer mobile

Commit:
created a global array workingRecipes to use in my filter and sort methods

Commit:
added random button + action to button

Commit:
added the math formula for indexes + console log

Commit:
added slice method to get my random selection to work

Commit:
added styling to random button

Commit:
added comments to every code snippet in js + a js.md

Commit:
updated the random math method + changed placement

Commit:
updated and finalized all comments

Commit:
updated comments on code snippets in js, final

Commit:
deleted some comments, to keep it clean

Commit:
added styling to surprise me button

Commit:
added code to fetch 1 random recipe from API, response logged in console

Commit:
added a fetch-button, called my function and saving API in variabel

Commit:
updated ingredient list in template literal w. ternary operator

Commit:
updated diet with cond.statements

Commit:
changed ingredients to cond.statement

Commit:
changed function ingredients from for.Each to map+join again

Commit:
updated a global base URL and changed function at the bottom of page

Commit:
made copy of baseURL and updated filter code w. if statement and fetch code

Commit:
removed prior filter code for static array

Commit:
updated sort code and removed old

Commit:
removed static array + global array

Commit:
removed random button and fetch button tat used static array from html, css

Commit:
removed old empty message code + added comments for each code snippet in js

Commit:
updated baseurl, apikey on top of page

Commit:
updated error message to match material

Commit:
added random button + styling + adapting randombutton to API

Commit:
added code in .then section to make all recipes show when page loads = fetchrecipe function

Commit:
removed non-veg button as it is all button

Commit:
js comments + console log, final

Commit:
removed all console log + updated comments for every code snippet, finalfinal

Commit:
removed alla comments for cleaner look

Commit:
moved copy of static url to top of page

Commit:
updated button style css + js diet info

Commit:
ne button with empty message

Commit:
updated function for pageload

Commit:
finalized html, css and js

Commit:
updated JS fetch recipe function with local storage logic + sort and filter w. local storage

govargas commented on index.html:
```diff
@@ -0,0 +1,115 @@
+    <h1>Recipe Library</h1>
+
+    <button id="button">Surprise me!</button>
+    <button id="button1">Chefs Choice</button>
```
The "Chef’s Choice" option is a nice personal touch on the project. Would it be like an own selection of recipes that you put as arrays? Just curious, as I think it is a good idea.

JasminHed replied:
Glad you like it! That is a good idea, perhaps I will implement it, for now I thought it would be the "empty message" button. I really wanted to see how such a message would be displayed.

govargas commented on styles.css:
The overall styling is clean and easy to read for me as a beginner. Thank you for that!

govargas commented on script.js:
```diff
@@ -0,0 +1,156 @@
+  const URL = "https://api.spoonacular.com/recipes/random?apiKey=f22e66767aca4bdda6a1d293d962b29e&number=10"
+  const recipesContainer = document.getElementById("recipes-container")
+
+  // Fetch API recipes, save them to local storage and when limit is hit show stored recipes + show an error message
+  const fetchRecipe = () => {
+    fetch(URL)
+      .then(response => {
+        if (!response.ok) {
+          recipesContainer.innerHTML = `<p>API limit reached, but don't worry! We've got a full load of saved recipes for you.</p>`
+          throw new Error("API limit reached")
+        }
+        return response.json()
+      })
+      .then(data => {
+        if (data && data.recipes && data.recipes.length > 0) {
+          localStorage.setItem("recipes", JSON.stringify(data.recipes)) // Save fetched recipes to localStorage.
+          loadRecipes(data.recipes)
+        }
+      })
+      .catch(() => {
+        const storedRecipes = localStorage.getItem("recipes")
+        if (storedRecipes) {
+          loadRecipes(JSON.parse(storedRecipes)) // parse means converting string from storage to a usabel JS array
+        }
+      })
+  }
```
The code seems to properly checks for API limits and falls back to localStorage. Nice! Interesting for me to see how it is implemented using promise chains (.then and .catch), as I went for async/await. Will try this next time.

JasminHed replied:
Yes, glad it shows! I kept running out of recipes so challenged myself to do a local storage. Happy I did. I will try async/await next time...

govargas commented on script.js:
```diff
@@ -0,0 +1,156 @@
+    surpriseButton.addEventListener("click", getRandomRecipe)
+  }
+
+  //Empty message
+  const getChefsChoice = () => {
+    recipesContainer.innerHTML = `<p>No recipes found today from our chef. Try another button!</p>`
+  }
```
I was a bit worried when I didn't got a Chef's Choice, but now I'm guessing that is on purpose in order to show a message to the user?

JasminHed replied:
Yes! hihi

govargas reviewed (commented):
Good job Jasmin! The layout is logical, well thought, easy to understand and navigate through. And the code seems to work smoothly. Everything is there for a reason, and I like you even added a “Chef’s Choice” option, although I didn’t manage to make it work. But maybe that was the intention, to use it in order to lead to an empty state and show a message to the user as part of the requirements? 

Just a small thing, but the link on the comment does not work. No biggie, as the link on the readme works perfectly.

Overall a nice, clean website. The code seems to meet the project requirements and demonstrates solid understanding. Great work!

HIPPIEKICK reviewed (approved):
Well named functions, keep this up Jasmin! If you get some time over, I’d love if you’d work some more on the CSS. I can see you’ve followed a lot of the Figma design which is great - but I think it would do soo much for the visuals if you’d add some paddings for your buttons and change the font to a sans-serif one (like Montserrat or similar).

You’ve reached all the requirements, so approving 💃

JasminHed commented:
Hello Matilda, 
Thanks, I think understanding the functions made all the difference :). I have now updated the font style and added some padding, just enough to make it all fit! Also might have gone too crazy on the colors but I changed them completely to more "me". Hope this makes a difference!

Commit:
updated style for buttons, font + color

Commit:
updated colors

HIPPIEKICK commented:
Wow, I think it looks super 🥳

Commit:
updated color that follow accessability

Commit:
better contrast AAA

Commit:
adjusted mediaq 480px

Commit:
updated mediaqueries 480px

Commit:
new colors

Commit:
tabbing+focus

Commit:
updated large px to rem

JasminHed committed:
Update README.md

JasminHed committed:
added new comments in js, removed old comments in css, added readme

JasminHed committed:
solved conflict

JasminHed committed:
updated readmefile

JasminHed committed:
updated alt, no repeat image text
